atomic_reactor/utils/imageutil.py

```diff
@@ -92,33 +100,71 @@ def _get_registry_client(self, registry: str) -> util.RegistryClient:
+                             that will be extracted
+        :param dst_path: str, path where to export file/dir
+        """
+        cmd = ['oc', 'image', 'extract', f'{image}', '--path', f'{src_path}:{dst_path}']
 
-def remove_image(image, force=False):
-    # self.tasker.remove_image(image, force=force)
-    # most likely won't be needed at all
-    return {}
+        try:
+            return subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image file extraction failed\n%s", e.output)
+            raise
 
+    def download_image_archive_tarball(self, image: str, path: str) -> None:
+        """Downloads image archive tarball to path.
 
-def tag_image(image, new_image):
-    # self.tasker.tag_image(image, new_image)
-    return True
+        :param image: str, image pullspec to download
+        :param path: str, path including the filename of the tarball
+        """
+        cmd = ['skopeo', 'copy', f'docker://{image}', f'docker-archive://{path}']
+        try:
+            retries.run_cmd(cmd)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image archive download failed:\n%s", e.output)
+            raise
+
+    def get_uncompressed_image_layer_sizes(self, path: str) -> List[Dict[str, Union[str, int]]]:
+        """Returns data about the uncompressed image layer sizes
+
+        :param path: str path to a image archive tarball
+        :returns Dict[str, int], where key is layer id and int is the size
+                                 in bytes
+        """
+        with tarfile.open(path) as tar:
+            manifest = json.load(tar.extractfile("manifest.json"))
+            layers = manifest[0]['Layers']
+            layer_sizes = []
+            for layer in layers:
+                layer_id = f'sha256:{layer.replace(".tar", "")}'
+                layer_sizes.append({
+                    "diff_id": layer_id,
+                    "size": tar.getmember(layer).size
+                })
+            return layer_sizes
+
+    def extract_filesystem_layer(self, src_path: str, dst_path: str) -> str:
+        """Extract filesystem layer from image archive tarball and
+        saves it at dst_path. This is meant for flatpacks and will work
+        only when the archive has 1 layer.
+
+        :param src_path: str, path to image archive tarball
+        :param dst_path: str, path where the layer will be copied
+
+        :returns str, the filesystem layer name
+        """
+        with tarfile.open(src_path) as tar:
+            manifest = json.load(tar.extractfile("manifest.json"))
+            layers = manifest[0]['Layers']
+            if len(layers) > 1:
+                raise ValueError(f'Tarball at {src_path} has more than 1 layer')
 
+            layer_file = tar.getmember(layers[0])
+            tar.extract(layer_file, dst_path)
 
-def get_image(image):
```

Comment:
@acmiel post_compress.py calls this method, is `download_image_archive_tarball` doing the same? or should this method do something different?

Reply:
I think the only difference is that `download_image_archive_tarball` creates a file on disk while the current usage assumes that `get_image` returns an open binary file. We don't need to keep `get_image` for that, just open the downloaded tarball instead

Reply:
Thanks, I updated the todo comment in post_compress.py and removed the call to get_image